Frends.AmazonS3.ListObjects/Frends.AmazonS3.ListObjects/ListObjects.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using System.ComponentModel;
+using System.Threading;
+using Frends.AmazonS3.ListObjects.Definitions;
+using System.Threading.Tasks;
+using Amazon;
+using Amazon.S3;
+using System.Collections.Generic;
+using Amazon.S3.Model;
+
+namespace Frends.AmazonS3.ListObjects
+{
+    /// <summary>
+    /// Amazon S3 task.
+    /// </summary>
+    public class AmazonS3
+    {
+        /// <summary>
+        /// List objects in AWS S3.
+        /// [Documentation](https://tasks.frends.com/tasks/frends-tasks/Frends.AmazonS3.ListObjects)
+        /// </summary>
+        /// <param name="source">Source connection parameters</param>
+        /// <param name="options">Options for the task</param>
+        /// <param name="cancellationToken">Token to stop ListObjects. This is generated by Frends.</param>
+        /// <returns>List { string BucketName, string Key, string Etag, long Size, DateTime LastModified  }</returns>
+        public static async Task<Result> ListObjects([PropertyTab] Source source, [PropertyTab] Options options, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrWhiteSpace(source.AwsSecretAccessKey) || string.IsNullOrWhiteSpace(source.AwsAccessKeyId))
+                throw new Exception("AWS credentials missing.");
+
+            var region = RegionSelection(source.Region);
+            var client = new AmazonS3Client(source.AwsAccessKeyId, source.AwsSecretAccessKey, region);
+            var response = await ListBucketContentsAsync(client, source, options, cancellationToken);
+            return new Result(response);
+        }
+
+#pragma warning disable CS1591 // Helping methods. No need for XML.
+        public static async Task<List<BucketObject>> ListBucketContentsAsync(AmazonS3Client client, Source source, Options options, CancellationToken cancellationToken)
```

Comment:
Since this is a helper method, it should be private. Also, you shouldn't need pragma if the method is private.